### ISSUE TITLE
Using regex in fos_js_routing.routes_to_expose no longer works correctly in 2.3.0

### ISSUE THREAD

**Issue by kriswillis** [FriendsOfSymfony/FOSJsRoutingBundle#355]
Since updating from 2.2.2 to 2.3.0, regex defined in `fos_js_routing.routes_to_expose` within `config.yml` are no longer matching our routes.

We're using the following patterns:

````
fos_js_routing:
    routes_to_expose: ["_api_", "_modal$", "_list$"]
````

Symfony 3.4.22
JsRoutingBundle 2.3.0

Rolling back to 2.2.2 exposes the routes correctly again.

**Comment by krixon** [comment]
Same here, this seems to have been introduced in ce741512474c654d703833d263d337da28e09fe1

```
protected function getDomainByRouteMatches($matches, $name)
    {
        $matches = array_filter($matches, function($match) {
            return !empty($match);
        });

        $matches = array_flip(array_intersect_key($matches, array_flip($this->availableDomains)));

        return isset($matches[$name]) ? $matches[$name] : null;
    }
```

Here `$name` is the full route name, but $matches contains the configured regex pattern so this always returns null.

**Comment by tobias-93** [comment]
Fixed using #350
